sdv/single_table/copulas.py

```diff
@@ -84,6 +86,25 @@ def __init__(self, metadata, enforce_min_max_values=True, enforce_rounding=True,
             for field, distribution in (numerical_distributions or {}).items()
         }
 
+    def _fit(self, processed_data):
+        """Fit the model to the table.
+
+        Args:
+            processed_data (pandas.DataFrame):
+                Data to be learned.
+        """
+        numerical_distributions = deepcopy(self._numerical_distributions)
```

Comment:
why do we have to copy here?

Reply:
I think that copy is good as we alter the `numerical_distributions` when fitting, this can allow us to re-fit the same instance with the same numerical distributions that we set prior (as you can see further in the code the `numerical_distributions` get updated, so just for sanity I wanted to use a copy.

Reply:
The thing is we are going to in the future expose methods that let users see the fitted distributions. As long as there is a way to retrieve them from the column name I think it's fine